analysis/cox_model.do

```diff
@@ -0,0 +1,217 @@
+lab val cov_cat_deprivation_tmp cov_cat_deprivation_tmp
+drop cov_cat_deprivation
+rename cov_cat_deprivation_tmp cov_cat_deprivation
+
+* Recode smoking status
+
+gen cov_cat_smoking_status_tmp = .
+replace cov_cat_smoking_status_tmp = 1 if cov_cat_smoking_status=="Never smoker"
+replace cov_cat_smoking_status_tmp = 2 if cov_cat_smoking_status=="Ever smoker"
+replace cov_cat_smoking_status_tmp = 3 if cov_cat_smoking_status=="Current smoker"
+replace cov_cat_smoking_status_tmp = 4 if cov_cat_smoking_status=="Missing"
+lab def cov_cat_smoking_status_tmp 1 "Never smoker" 2 "Ever smoker" 3 "Current smoker" 4 "Missing"
+lab val cov_cat_smoking_status_tmp cov_cat_smoking_status_tmp
+drop cov_cat_smoking_status
+rename cov_cat_smoking_status_tmp cov_cat_smoking_status 
+
+* Summarize missingness following recoding
+
+misstable summarize
+	
+* Make failure variable
+
+gen outcome_status = 0
+replace outcome_status = 1 if outcome_date!=.
+
+* Update follow-up end
+
+replace follow_up_end = follow_up_end + 1
+format follow_up_end %td
+
+* Make age spline
+
+centile age, centile(10 50 90)
+mkspline age_spline = age, cubic knots(`r(c_1)' `r(c_2)' `r(c_3)')
+
+* Apply stset // including IPW here as if unsampled dataset will be 1
+
+stset follow_up_end [pweight=cox_weights], failure(outcome_status) id(patient_id) enter(follow_up_start) origin(time mdy(06,01,2021))
+stsplit days, after(exposure_date) at(0 28 197)
+
+* Calculate study follow up
+
+replace days = 535 if days==-1 & prevac_cohort==1 // prevaccination cohort
+replace days = 197 if days==-1 & prevac_cohort!=1 // vac & unvac cohorts
+gen follow_up = _t - _t0
+egen follow_up_total = total(follow_up)  
+
+* Make days variables
+
+gen days0_28 = 0
+replace days0_28 = 1 if days==0
+tab days0_28
+
+gen days28_197 = 0
+replace days28_197 = 1 if days==28
+tab days28_197
+
+gen days197_535 = 0 & prevac_cohort==1 // prevaccination cohort only
+replace days197_535 = 1 if days==197 & prevac_cohort==1
+tab days197_535
+
+* Run models and save output [Note: cannot use efron method with weights]
+
+tab days outcome_status 
+
+di "Total follow-up in days: " follow_up_total
+bysort days: summarize(follow_up), detail
+
+stcox days* i.sex age_spline1 age_spline2, strata(region) vce(r)
+est store min, title(Age_Sex)
+stcox days* i.sex age_spline1 age_spline2 i.cov_cat_ethnicity i.cov_cat_deprivation i.cov_cat_smoking_status cov_num_consulation_rate cov_bin_*, strata(region) vce(r)
+est store max, title(Maximal)
+
+estout * using "output/`cpf'_cox_model.txt", cells("b se t ci_l ci_u p") stats(risk N_fail N_sub N N_clust) replace 
+
+* Calculate median follow-up
+
+keep if outcome_status==1
+drop if days0_28==0 & days28_197==0
+keep patient_id days0_28 days28_197 follow_up
+
+gen term = ""
+replace term = "days0_28" if days0_28==1 & days28_197==0
+replace term = "days28_197" if days0_28==0 & days28_197==1
+
+replace follow_up = follow_up + 28 if term == "days28_197"
```

Comment:
Also, add the following line:
`replace follow_up = follow_up + 197 if term == "days197_535"`